Double click to place preset

squidmade opened the pull request:
This change allows placing a preset by double clicking its entry in the library window, useful for those who place presets directly from the plugin UI instead of the game's waymarks window.

While this makes it easier to place a single preset, it also makes it quicker to switch between presets. A use case for this is in a savage PF group (on NA) when assigning positions and roles before the first pull. Because NA unfortunately doesn't use macros, this setup often involves multiple waymark configurations as shown in the gif below:

![double_click_to_place_preset](https://user-images.githubusercontent.com/99750849/178594093-1857bf2a-6d86-41fb-b11c-0dc9ebc9ec56.gif)

For testing, I made sure that this change doesn't interact adversely with click to unselect preset turned on.

squidmade committed:
In the library window, implemented double clicking a preset to place waymarks.

squidmade committed:
Added check to prevent placing an invalid preset when double clicking and click to unselect preset is enabled.

PunishedPineapple commented:
Thanks!  This is on the list to add in with some context menu stuff when I get a chance to work on the plugin again and do another release.

~~(Note to self: Test with editor window open and make sure that it can't affect the selection in that case.)~~

squidmade committed:
Replaced deprecated calls to GetWindowContentRegionWidth() to fix compile errors with the new version of ImGui

squidmade committed:
Prevent unselect when a preset is double clicked and AllowUnselectPreset is enabled. Single click still unselects the preset.

squidmade commented:
I fixed compile errors caused by deprecated function calls as a result of the ImGui version bump. I also fixed an issue when AllowUnselectPreset is enabled where Preset Info would show and then hide when double clicked. It wasn't breaking but looked weird:

![waymark-show-hide](https://user-images.githubusercontent.com/99750849/179091553-3ef52015-9cb3-421a-a8cb-9f330d395dac.gif)

Using the ImGuiSelectableFlags.AllowDoubleClick flag, this version more properly checks for the double click inside of the Selectable()'s if-statement. Behavior with the fix applied:

![waymark-fixed](https://user-images.githubusercontent.com/99750849/179093589-59281dcb-1cc3-4c25-a6f1-0636e6d39c4b.gif)

There's still a minor Preset Info hide/show issue when an already selected preset is double clicked. I did some debug logging and saw that the unselect and select are happening on separate frames, so to fix this I believe a timer is needed. I can do that if you'd like this fixed as well.

![waymark-hide-show](https://user-images.githubusercontent.com/99750849/179096605-9b7c7ca4-fb10-4a96-9048-c6292c84a34b.gif)

PunishedPineapple commented:
Where you replaced `ImGui.GetWindowContentRegionWidth()` with `ImGui.GetWindowContentRegionMax().X`, it should be `( ImGui.GetWindowContentRegionMax().X - ImGui.GetWindowContentRegionMin().X )` (parentheses where required by order of operations) or `ImGui.GetContentRegionAvail().X` (I actually haven't tested the latter).  If you want to fix it in your PR, go ahead; otherwise I'll just fix it post-merge.  I'll merge and do a release tonight after raid.
